### PR TITLE
update to latest nightly version

### PR DESCRIPTION
**Pull request by Velfi**
This PR updates the logic around how hidden modules and items are processed.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by Velfi** [comment]
Man this is a tough one.

File: src/visitor.rs [hunk 1]
```diff
@@ -641,8 +659,10 @@ impl Visitor {
         }
         // Crates like `pin_project` do some shenanigans to create and reference types that don't end up
         // in the doc index, but that should only happen within the root crate.
-        else if !id.0.starts_with(&format!("{}:", self.root_crate_id)) {
-            unreachable!("A type is referencing another type that is not in the index, and that type is from another crate.");
+        else if self.in_external_crate(id) {
+            unreachable!(
+                "An item referenced another item from an external crate that is not in the index."
+            );
         }
```
Comment (Velfi) [inline src/visitor.rs:666]: I still have to figure out how to handle the docs visitor hitting this.

Comment (Velfi) [reply src/visitor.rs]: OK, the issue was that we were choking on `doc(hidden)` items. I added a new kind of warning for those.

File: src/visitor.rs [hunk 2]
```diff
@@ -219,14 +231,16 @@ impl Visitor {
...
                     // and we don't want to examine the innards of the re-export.
                     if module_item.crate_id == self.root_crate_id {
-                        self.visit_item(&path, module_item, VisibilityCheck::Default).context(here!())?;
+                        self.visit_item(&path, module_item, VisibilityCheck::Default)
+                            .context(here!())?;
                     }
                 }
             }
-            ItemEnum::OpaqueTy(_) => unstable_rust_feature!("type_alias_impl_trait", "https://doc.rust-lang.org/beta/unstable-book/language-features/type-alias-impl-trait.html"),
+            // ItemEnum::OpaqueTy(_) => unstable_rust_feature!("type_alias_impl_trait", "https://doc.rust-lang.org/beta/unstable-book/language-features/type-alias-impl-trait.html"),
```
Comment (landonxjames) [inline src/visitor.rs:239]: Any reason to keep this commented out?

**Review by jdisanti — APPROVED** [review]
Nice!

**Review by landonxjames — APPROVED** [review]
LGTM

**Pull request merged**